Spotkanie5_Zadania/CurrencyConversion_v2/CurrencyConversion/Transaction.cs

```diff
@@ -0,0 +1,81 @@
+﻿namespace CurrencyConversion
+{
+    class Transaction
+    {
+        AppOption appOption = new AppOption();
+        Cantor cantor = new Cantor();
+        public decimal amount;
+        public void BuyOrSell()
+        {
+            Console.Write("Dear user, tell me what you want? Buy?\n");
+            int buy = appOption.DecideWhichOption();
+            if (buy == 1) Buying();
+            else if (buy == 2) Selling();
+        }
+
+        public void Buying()
+        {           
+            if (OwnCurrency() == 2)
+            {
+                Console.Write("User, to sure do you want to use ready-made currencies?: \n");
+                int assignItYourself = appOption.DecideWhichOption();
+
+                if (assignItYourself == 1)
+                {
+                    var result = cantor.PurchaseCurrency();
+                    string currency = result.Item1;
+                    decimal currencyValue = Math.Round(result.Item2, 4);
+                    currencyValue = Math.Round(currencyValue, 4);
+                    Console.Write($"I understand my dear user, you have chosen {currency}, and how many {currency}s do you need buy: ");
+                    amount = appOption.DownloadAmount();
+                    Console.WriteLine($"{amount} {currency}s will cost you: {Math.Round(amount * currencyValue, 2)} PLN.");
+                }
+                else if (assignItYourself == 2) appOption.NoDecisionYet();
+            }
+        }
+
+        public void Selling()
```

Comment:
Sporo zagniezdzen, ifow i else'ow w tej metodzie. Sugeruje jeszcze raz przemyslec i sprobowac to rozplatac w miare mozliwosci

Reply:
Ty fakytycznie xD